engines/sherlock/user_interface.cpp

```diff
@@ -0,0 +1,2379 @@
+	"Nothing of interest here", "It is bolted down", "It is too big to carry", "It is too heavy",
+	"I think a girl would be more your type", "Those flowers belong to Penny", "She's far too young for you!",
+	"I think a girl would be more your type!", "Government property for official use only"
+};
+const char *const MUSE[] = {
+	"You can't do that", "It had no effect", "You can't reach it", "OK, the door looks bigger! Happy?",
+	"Doors don't smoke"
+};
+
+/*----------------------------------------------------------------*/
+
+UserInterface::UserInterface(SherlockEngine *vm) : _vm(vm) {
+	_controls = new ImageFile("menu.all");
+	_controlPanel = new ImageFile("controls.vgs");
+	_bgFound = 0;
+	_oldBgFound = -1;
+	_keycode = Common::KEYCODE_INVALID;
+	_helpStyle = false;
+	_menuCounter = 0;
+	_menuMode = STD_MODE;
+	_help = _oldHelp = 0;
+	_lookHelp = 0;
+	_key = _oldKey = 0;
+	_temp = _oldTemp = 0;
+	_temp1 = 0;
+	_invLookFlag = 0;
+	_windowOpen = false;
+	_oldLook = false;
+	_keyboardInput = false;
+	_pause = false;
+	_cNum = 0;
+	_selector = _oldSelector = -1;
+	_windowBounds = Common::Rect(0, CONTROLS_Y1, SHERLOCK_SCREEN_WIDTH - 1,
+		SHERLOCK_SCREEN_HEIGHT - 1);
+	_windowStyle = 1;	// Sliding windows
+	_find = 0;
+	_oldUse = 0;
+	_endKeyActive = true;
+	_lookScriptFlag = false;
+	_infoFlag = false;
+}
+
+UserInterface::~UserInterface() {
+	delete _controls;
+	delete _controlPanel;
+}
+
+/**
+ * Resets the user interface
+ */
+void UserInterface::reset() {
+	_oldKey = -1;
+	_help = _oldHelp = -1;
+	_oldTemp = _temp = -1;
+}
+
+/**
+ * Draw the user interface onto the screen's back buffers
+ */
+void UserInterface::drawInterface(int bufferNum) {
+	Screen &screen = *_vm->_screen;
+
+	if (bufferNum & 1)
+		screen._backBuffer1.transBlitFrom((*_controlPanel)[0], Common::Point(0, CONTROLS_Y));
+	if (bufferNum & 2)
+		screen._backBuffer2.transBlitFrom((*_controlPanel)[0], Common::Point(0, CONTROLS_Y));
+	if (bufferNum == 3)
+		screen._backBuffer2.fillRect(0, INFO_LINE, SHERLOCK_SCREEN_WIDTH, INFO_LINE + 10, INFO_BLACK);
+}
+
+/**
+ * Main input handler for the user interface
+ */
+void UserInterface::handleInput() {
+	Events &events = *_vm->_events;
+	Inventory &inv = *_vm->_inventory;
+	People &people = *_vm->_people;
+	Scene &scene = *_vm->_scene;
+	Screen &screen = *_vm->_screen;
+	Talk &talk = *_vm->_talk;
+
+	if (_menuCounter)
+		whileMenuCounter();
+
+	Common::Point pt = events.mousePos();
+	_bgFound = scene.findBgShape(Common::Rect(pt.x, pt.y, pt.x + 1, pt.y + 1));
+	_keycode = Common::KEYCODE_INVALID;
+
+	// Check kbd and set the mouse released flag if Enter or space is pressed.
+	// Otherwise, the pressed _key is stored for later use
+	if (events.kbHit()) {
+		Common::KeyState keyState = events.getKey();
+		_keycode = keyState.ascii;
+
+		if (keyState.keycode == Common::KEYCODE_x && keyState.flags & Common::KBD_ALT) {
+			_vm->quitGame();
+			events.pollEvents();
+			return;
+		} else if (keyState.keycode == Common::KEYCODE_SPACE ||
```

Comment:
I don't think it is; at least, I tried the prompts with other keys and they work fine. This block is simply special handling to make click checks work with the keyboard as well. Honestly, I'm not even sure if it's needed.. The original technically had a "keyboard cursor" that could be controlled via the number pad, but we chose not to implement it. I think that this is a leftover fragment. Although it certainly doesn't hurt leaving it in place.